react/features/recording/components/Recording/AbstractStartRecordingDialog.js

```diff
@@ -107,9 +113,29 @@ class AbstractStartRecordingDialog extends Component<Props, State> {
      * @returns {void}
      */
     componentDidMount() {
-        if (typeof this.props._token !== 'undefined') {
+        const {
+            _token,
+            _fileRecordingsServiceEnabled
+        } = this.props;
+
+        if (typeof _token !== 'undefined') {
             this._onTokenUpdated();
         }
+
+        // FIXME: Fix the eslint warning. Potentially check if we need to handle
+        // changes of _fileRecordingsServiceEnabled and
+        // _areIntegrationsEnabled()
+        if (_fileRecordingsServiceEnabled || !this._areIntegrationsEnabled()) {
+            // eslint-disable-next-line react/no-did-mount-set-state
+            this.setState({
+                selectedRecordingService: RECORDING_TYPES.JITSI_REC_SERVICE
```

Comment:
You could do this check in the constructor set the state then, avoiding the eslint disabling.

Reply:
Good catch! It's fixed!